src/components/RuleItem.vue

```diff
@@ -188,24 +263,72 @@ export default {
       Vue.delete(ruleParent, containerName);
       this.$emit('followLink', parentPath);
     },
+
+    // constraints
     addConstraint() {
       if (!this.constraints) {
         Vue.set(this.rule, 'constraint', []);
       }
+
+      // make use of the logical operator to combine more than one constraint
+      if (this.constraints.length == 1) {
+        const constraint = this.constraints;
+        Vue.delete(this.rule, 'constraint');
+        Vue.set(this.rule, 'constraint', {});
+        Vue.set(this.rule.constraint, this.logicalConOpShort, {});
+        Vue.set(this.rule.constraint[this.logicalConOpShort], '@list', constraint);
```

Comment:
test whether we can construct the object literally and then start watching it with only one call to Vue.set?